examples/defuns.kl

```diff
@@ -0,0 +1,261 @@
+#lang "prelude.kl"
+(import (shift "prelude.kl" 1))
+(import (shift "do.kl" 1))
+(import "either-datatype.kl")
+(import "list.kl")
+(import (shift "list.kl" 1))
+(import "pair-datatype.kl")
+(import (shift "pair-datatype.kl" 1))
+
+-- (defuns
+--   [even? (n)
+--    (if (= n 0) true (odd? (- n 1)))]
+--   [odd? (n)
+--    (if (= n 0) false (even? (- n 1)))])
+-- =>
+-- (group
+--   (defun even?/odd? (n/n)
+--     (flet [even? (n)
+--            (case (even?/odd? (left n))
+--              [(left r)
+--               r]
+--              [_
+--               (error '"never happens: even?/odd? always returns a left when receiving a left")])]
+--       (flet [odd? (n)
+--              (case (even?/odd? (right n))
+--                [(right r)
+--                 r]
+--                [_
+--                 (error '"never happens: even?/odd? always returns a right when receiving a right")])]
+--         (case n/n
+--           [(left n)
+--            (left (if (= n 0) true (odd? (- n 1))))]
+--           [(right n)
+--            (right (if (= n 0) false (even? (- n 1))))]))))
+--   (defun even? (n)
+--     (case (even?/odd? (left n))
+--       [(left r)
+--        r]
+--       [_
+--        (error '"never happens: even?/odd? always returns a left when receiving a left")]))
+--   (defun odd? (n)
+--     (case (even?/odd? (right n))
+--       [(right r)
+--        r]
+--       [_
+--        (error '"never happens: even?/odd? always returns a right when receiving a right")])))
+(define-macros
+  ([defuns
+    -- (parse-def '[even? (n)
+    --              (if (= n 0) true (odd? (- n 1)))])
+    -- =>
+    -- (pair 'even?
+    --       '(lambda (n) (if (= n 0) true (odd? (- n 1)))))
+    (flet [parse-def (stx)
+           (syntax-case stx
+             [(list (fN-name args body))
+              (syntax-case args
+                [()
```

Comment:
Just an observation: coming from Racket, this seems a bit weird to me. In Racket, I would write either:

```
(match v
  [(nil) ...]
  [(cons x xs) ...])
```

or

```
(syntax-case v
  [() ...]
  [(x xs ...) ...])
```

It's unclear if `syntax-case` in Klister is meant to be more like `match` or `syntax-case` in Racket. 

If former, I would expect the first branch to be `(nil)`, so that pattern matching and constructor mirror each other. 

If latter, I would expect `cons` to not be there (or rather, if it's there, it would be a pattern matching against a list of length three, with `cons`, `first-arg`, and `more-args` as three pattern variables).

Reply:
Thanks for the suggestion, `(nil)` would indeed be clearer.

Reply:
Fixed in subsequent PR #169